src/core/tools/common/network.sim.cpp

```diff
@@ -91,12 +91,8 @@ namespace dsn { namespace tools {
     {
         for (auto& msg : _sending_msgs)
         {
-            network_header_format hdr_format(NET_HDR_DSN);
-            bool r = header_type::header_type_to_format(msg->header->hdr_type, hdr_format);
-            dassert(r, "header_type_to_format(%s) failed", msg->header->hdr_type.debug_string().c_str());
-
             sim_network_provider* rnet = nullptr;
-            if (!s_switch[task_spec::get(msg->local_rpc_code)->rpc_call_channel][hdr_format].get(remote_address(), rnet))
+            if (!s_switch[task_spec::get(msg->local_rpc_code)->rpc_call_channel][NET_HDR_DSN].get(remote_address(), rnet))
```

Comment:
the hdr_format should be determined by "msg->header->hdr_type", but not NET_HDR_DSN const value